### PR TITLE
wifi: eswifi: fix random crash due to work->handler set to NULL

### PR DESCRIPTION
**Pull request by mzajac-avs**
This fixes the random crash on L475 disco iot1 board described here: https://github.com/zephyrproject-rtos/zephyr/issues/49963

### CODE REVIEW

File: drivers/wifi/eswifi/eswifi_offload.c [hunk 1]
```diff
@@ -397,7 +397,7 @@ static int eswifi_off_put(struct net_context *context)
 	}
 
 	if (--socket->usage <= 0) {
-		memset(socket, 0, sizeof(*socket));
+		socket->context = NULL;
```
Comment (loicpoulain) [inline drivers/wifi/eswifi/eswifi_offload.c:400]: That certainly works, but wouldn't it be possible to cancel any ongoing/scheduled work instead? I mean once socket is 'released' it's no more assigned to a context, and can be reused at any time, so we don't want anything to access it anymore after this line.

Comment (mzajac-avs) [reply drivers/wifi/eswifi/eswifi_offload.c:400]: Well I'm not sure did I try to do this correctly but I added a `static struct k_work_sync work_sync` to the file and called `        k_work_cancel_delayable_sync(&socket->read_work, &work_sync);` before the `memset()`. I tired it a few times but it seems that the execution of the application frozen each time (but there was no crash). The stack showed that I'm in `arch_cpu_idle () at /home/mzajac/Repo/Internal/zephyr-root/zephyr/arch/arm/core/aarch32/cpu_idle.S:105`

Comment (loicpoulain) [reply drivers/wifi/eswifi/eswifi_offload.c:400]: yes, though it's more a workaround than a fix. but better than nothing.

**Review by loicpoulain — DISMISSED** [review]

**Review by nandojve — DISMISSED** [review]

**Comment by erwango** [comment]
@mzajac-avs Please check your commit message. It should conform to the following template:
```
<dir>: <dir>: <Title summarizing the change>

<A description useful to the reader>

Signed-off-by: foo <foo@bar.baz>
```

For references, I'd advise you to check:
- [Contribution guidelines on docs.zephyrproject.org](https://docs.zephyrproject.org/latest/contribute/guidelines.html)
- [Great presentation dealing with "the best way to get your patches merged"](https://youtu.be/mnVNLM7ynMQ) and the related [Slides](https://static.sched.com/hosted_files/zephyr2022/a3/HOWTO_Get_Your_Zephyr_Patches_Merged_ZDS_2022.pdf)
- Coding Guidelines bot results reported below.

**Review by erwango — APPROVED** [review]

**Comment by erwango** [comment]
@loicpoulain, @nandojve can you approve again? Your approval got lost.

**Review by loicpoulain — APPROVED** [review]

**Review by nandojve — APPROVED** [review]

**Pull request merged**